### DISCUSSION TITLE
Need to improve this library with JavaScript

### DISCUSSION THREAD

**Discussion by javidabedian** [saadeghi/daisyui#1373] [General]
Hello there , thank you for built this amazing library Pouya @saadeghi 

The pre-built css component looks so good with this library , But need to improve some of the components with JavaScript .

Need to add pre-defined JavaScript based component I was thinking about it :

- Navbar with dropdown or Navbar with user avatar dropdown
- Dropdown menu with Javascript
- Tabs with Javascript
- Alert include close button with Javascript
- Modal with Javascript

there is a problem with label and :active pseudo class , for example when I click on a label to show dropdown menu , I must click outside of the dropdown to make it hide again , But some of other libraries work with toggle class for fix this issue .

If there is a pre-defined JavaScript file to fix those issue and make this library more easy to use , it's gonna be so good .

Thank you again Pouya for this amazing library <3

**Comment by Purus** [comment]
I was trying to avoid using any other JS framework and if something is built into daisyUI that would be great.

**Reply by polyzen** [reply]
A repo with examples on how to enhance various daisyUI components using JS/frameworks could be nice.

**Comment by saadeghi** [comment]
There are lots of framework specific JS packages that you can easily use alongside daisyUI. For modal and dropdown you can use [Headless UI](https://headlessui.com/) (from Tailwind team) or [Radix UI](https://www.radix-ui.com/)  
And managing `tab` contents or closing `alert` is really easier to be done using your JS framework (depending on the syntax). It's just adding/removing class names or showing/hiding content when clicked.

There are also countless other JS packages for each framework that have no design decision (so you can use your own style for them or easily apply daisyUI/Tailwind class names)  
And I would suggest using the framework specific packages. Here's the reason:  
Let's say there's a JS bundle from daisyUI that handles modals (including the modal content, triggers, accessibility, keyboard navigation, etc). It should be vanilla JS and it should work on any framework. That would cost relatively more JS code shipped to the browser comparing to a framework specific package. For example if you're using React and the code to handle modal uses React states, it would be a less code shipped to the browser because React library is already bundled and shipped to the browser and you're just using it for your modal.  
It's the same scenario for other components as well. It's more efficient to use framework specific packages based on the framework you're using and even if you're not using JS frameworks at all, there are still vanilla JS plugins that can help you.  

The only efficient way for daisyUI to provide a JS bundle file would be providing multiple specific bundle files for each JS framework.  
But what would be the point of releasing new JS package when there are already existing JS packages for each framework that are already stable and can be used alongside daisyUI?

**Comment by polyzen** [comment]
TIL https://github.com/daisyui/react-daisyui
